pyqir/tests/test_if.py

```diff
@@ -1,431 +1,431 @@
-    qis = BasicQisBuilder(brancher.module.builder)
-
-    brancher.if_(
-        brancher.module.results[0],
-        true=lambda: qis.x(brancher.module.qubits[0]),
-        false=lambda: qis.h(brancher.module.qubits[0]),
-    )
-
-    logger = GateLogger()
-    _eval(brancher.module, logger)
-    assert logger.instructions == ["h qubit[0]"]
-
-
-def test_icmp_if_true() -> None:
-    brancher = _BoolBrancher(1)
-    x = brancher.oracle()
-    module = brancher.module
-    cond = module.builder.icmp(
-        IntPredicate.EQ, x, pyqir.const(IntType(module.context, 1), 0)
-    )
-
-    qis = BasicQisBuilder(module.builder)
-    brancher.if_(
-        cond,
-        lambda: qis.x(module.qubits[0]),
-        lambda: qis.h(module.qubits[0]),
-    )
-
-    logger = GateLogger()
-    _eval(brancher.module, logger, [False])
-    assert logger.instructions == ["m qubit[0] => out[0]", "x qubit[0]"]
-
-
-def test_icmp_if_false() -> None:
-    brancher = _BoolBrancher(1)
-    x = brancher.oracle()
-    module = brancher.module
-    cond = module.builder.icmp(
-        IntPredicate.EQ, x, pyqir.const(IntType(module.context, 1), 0)
-    )
-
-    qis = BasicQisBuilder(brancher.module.builder)
-    brancher.if_(
-        cond,
-        lambda: qis.x(module.qubits[0]),
-        lambda: qis.h(module.qubits[0]),
-    )
-
-    logger = GateLogger()
-    _eval(brancher.module, logger, [True])
-    assert logger.instructions == ["m qubit[0] => out[0]", "h qubit[0]"]
-
-
-@pytest.mark.parametrize("result", [True, False])
-def test_arithmetic_in_branch(result: bool) -> None:
-    brancher = _BoolBrancher(1)
-    cond = brancher.oracle()
-    module = brancher.module
-    qis = BasicQisBuilder(module.builder)
-    i32 = IntType(module.context, 32)
-
-    def true() -> None:
-        four = module.builder.add(pyqir.const(i32, 2), pyqir.const(i32, 2))
-        cond = module.builder.icmp(IntPredicate.EQ, four, pyqir.const(i32, 4))
-        module.builder.if_(cond, lambda: qis.x(module.qubits[0]))
-
-    brancher.if_(cond, true)
-
-    logger = GateLogger()
-    _eval(brancher.module, logger, [result])
-    expected = ["m qubit[0] => out[0]"] + (["x qubit[0]"] if result else [])
-    assert logger.instructions == expected
-
-
-@pytest.mark.parametrize("result", [True, False])
-def test_call_in_branch(result: bool) -> None:
-    brancher = _BoolBrancher(1)
-    module = brancher.module
-    context = module.context
-    x = module.add_external_function(
-        "__quantum__qis__x__body",
-        FunctionType(Type.void(context), [pyqir.qubit_type(context)]),
-    )
-
-    cond = brancher.oracle()
-
-    def apply_x_qubit0() -> None:
-        module.builder.call(x, [module.qubits[0]])
-
-    brancher.if_(cond, apply_x_qubit0)
-
-    logger = GateLogger()
-    _eval(brancher.module, logger, [result])
-    expected = ["m qubit[0] => out[0]"] + (["x qubit[0]"] if result else [])
-    assert logger.instructions == expected
+# # Copyright (c) Microsoft Corporation.
```

Comment:
Disabled tests. Some could be changed to test IR equality. I think there's a lot of overlap with the qirlib if tests now, though, so probably some could also be removed.